### PR TITLE
Converting console.error to throw error. #180

### LINKED ISSUE #180: Change silent errors to actually throw
Currently `angular-fontawesome` logs errors (search for `console.error`) instead of throwing them. The goal of this story is to always throw errors, so they are not lost.

### PR DESCRIPTION
**Pull request by plinkpaste**
Fixes #180

### CODE REVIEW

File: src/lib/icon/icon.component.ts [hunk 1]
```diff
@@ -134,9 +134,7 @@ export class FaIconComponent implements OnChanges {
       const message = 'Global icon library is deprecated. ' +
         'Consult https://github.com/FortAwesome/angular-fontawesome/blob/master/UPGRADING.md ' +
         'for the migration instructions.';
-      if (this.config.globalLibrary === 'unset') {
-        console.error('FontAwesome: ' + message);
-      } else if (!this.config.globalLibrary) {
+      if (!this.config.globalLibrary || this.config.globalLibrary === 'unset') {
```
Comment (devoto13) [inline src/lib/icon/icon.component.ts]: Please revert this change. 

`'unset'` value should behave the same as before: print warning. This option may be used by the consumers with the large codebases to opt-out of the hard error, while they update their applications.

Instead [change the default value](https://github.com/FortAwesome/angular-fontawesome/blob/master/src/lib/config.ts#L39) from `'unset'` to `false` in the `FaConfig` as described in the doc comment on that property.

**Comment by devoto13** [comment]
PS Also can you add text "Fixes #180" in the PR description, so the issue will be automatically closed, when PR is merged?

File: src/lib/icon/icon.component.spec.ts [hunk 2]
```diff
@@ -1,14 +1,14 @@
-import { Component, ComponentFactoryResolver, ViewChild, ViewContainerRef } from '@angular/core';
```
[the remark this answers is not in the document]
Comment (plinkpaste) [reply src/lib/icon/icon.component.spec.ts]: This was unintentional, and a case of the IDE being "helpful"! Reverted.

File: src/lib/icon/icon.component.ts (hunk 1 above)
Comment (plinkpaste) [reply src/lib/icon/icon.component.ts]: Understood. Reverted.

File: src/lib/config.ts [hunk 3]
```diff
@@ -36,5 +36,5 @@ export class FaConfig {
    *
    * @default 'unset'
```
Comment (devoto13) [inline src/lib/config.ts]: Please update the default value in JSDoc comment accordingly.

Comment (plinkpaste) [reply src/lib/config.ts]: Of course.

**Pull request merged**